daml-lf/spec/daml-lf-1.rst

```diff
@@ -680,6 +713,13 @@ Then we can define our kinds, types, and expressions::
   because erasable kinds have no runtime representation. This affects the
   operational semantics. The right hand side of an arrow is always erasable.
 
+.. note:: The explicit syntax for BigNumeric literal (case
+  ``ExpLitBigNumeric``) is forbidden in serialized programs. It is
```

Comment:
Why? I want to be able to write `\x -> x * 2.0` in Daml without going through a numeric conversion.

Reply:
This does not prevent you to have that `BigNumeric` literal in Daml.
 - The simplest is to limit DAML `BigNumeric` literals to those that can be represented as `Numerics`, and wrapped those in a `TO_BIGNUMERIC_NUMERIC`  when generated the LF code. I think this make complete sens as Numeric are not serializable; `BigNumeric` should be use only for intermediary computation. 
 - If you absolutely want that all valid `BigNumeric` have a valid DAML literal (what I really not sure we should) you can still internally construct them using `TO_BIGNUMERIC_NUMERIC`, `ADD_BIGNUMERIC` and `SHIFT_BIGNUMERIC` and cache the result in an LF  `val`.

Reply:
What do we gain from disallowing `BigNumeric` literals? I disagree that this is related to serializability. We clearly need literals in intermediate computations. I think you’re probably right that the numeric wrapper is sufficient but it seems worse from a performance pov with relatively little benefit.

Reply:
When I mean no literals, I mean there is not literal in the syntax i.e. in the archive, there is of course literal operationally. 
It is a bit like GenMap, we do not have literals in the syntax, you can create Map only bu using the builtin.

Reply:
Will try to reformulate